### PR TITLE
Add Codespell test to PRs

### PR DESCRIPTION
**Pull request by bandi13**
Adding in a spell checker

### CODE REVIEW

File: .github/workflows/coverity-scan-fixes.yml [hunk 1]
```diff
@@ -3,11 +3,7 @@ name: Coverity Scan master branch on a weekly basis
 on:
   workflow_dispatch:
   schedule:
-    - cron: "0 0,12 * * *"
```
Comment (philljj) [inline .github/workflows/coverity-scan-fixes.yml:6]: What is the coverity change for?

File: .github/workflows/codespell.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+    - uses: actions/checkout@v4
+
+    - uses: codespell-project/actions-codespell@v2.1
+      with:
+        check_filenames: true
+        check_hidden: true
+          # Add comma separated list of words to ignore (sorted alphabetically, case sensitive)
+        ignore_words_list: adin,ADn,alph,aNULL,carryIn,cconfiguration,chainG,ciph,cLen,cliKs,creen,dout,ede,Fo,haveA,hashIn,larg,LEAPYEAR,inCreated,inOut,inout,Merget,mot,optionA,parm,parms,repid,rIn,rsource,ser,siz,Te,te,TOOM,userA,vie
+        exclude_file: ./IDE/ECLIPSE/DEOS/tls_wolfssl.c,./IDE/ECLIPSE/MICRIUM/client_wolfssl.c,./IDE/IAR-MSP430/main.c,./tests/api.c
```
Comment (philljj) [inline .github/workflows/codespell.yml:27]: A comment about why we are excluding certain files would be nice.

File: .github/workflows/coverity-scan-fixes.yml
Comment (embhorn) [inline .github/workflows/coverity-scan-fixes.yml:1]: Please update the `name:` field to say daily instead of weekly

**Review by philljj — CHANGES_REQUESTED** [review]
We should skip files that throw codespell utf-8 warnings: `.csv`, `.der`, any Japanese translation txt files, etc.

Recommend merge squashing this to 1 commit.

File: .github/workflows/coverity-scan-fixes.yml (hunk 1 above)
Comment (bandi13) [reply .github/workflows/coverity-scan-fixes.yml:6]: @embhorn and I talked about this offline, that it makes sense to run it only once a day. Also fixed the comment to reflect it

File: .github/workflows/codespell.yml (hunk 2 above)
Comment (bandi13) [reply .github/workflows/codespell.yml]: Added comments.

**Review by douzzer — CHANGES_REQUESTED** [review]
alas we (I) merged a PR with a spelling error in it:
```
Error: ./wrapper/CSharp/wolfSSL_CSharp/wolfCrypt.cs:1633: Clenup ==> Cleanup
```
the sooner we get this PR merged, the better!

**Comment by bandi13** [comment]
retest this please

File: IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/test/test.scfg [hunk 3]
```diff
@@ -722,125 +722,125 @@
             <allocatable id="TMR0">
                 <isocket id="CountingClock" selection="Clock.tool_clock_pclkb" value="6.0E7"/>
                 <option enabled="true" id="CountMode" selection="8bitMode">
-                    <item id="8bitMode" input="" vlaue="0"/>
-                    <item id="16bitMode" input="" vlaue="0"/>
+                    <item id="8bitMode" input="" value="0"/>
```
Comment (dgarske) [inline IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/test/test.scfg:725]: Not sure you can/should fix these. This is a property for a code generation tool. Prefer you just add an exception for these files.

File: IDE/Renesas/e2studio/RX65N/GR-ROSE/smc/smc.scfg [hunk 4]
```diff
@@ -564,125 +564,125 @@
             <allocatable id="TMR0">
                 <isocket id="CountingClock" selection="Clock.tool_clock_pclkb" value="3.0E7"/>
                 <option enabled="true" id="CountMode" selection="8bitMode">
-                    <item id="8bitMode" input="" vlaue="0"/>
-                    <item id="16bitMode" input="" vlaue="0"/>
+                    <item id="8bitMode" input="" value="0"/>
```
Comment (dgarske) [inline IDE/Renesas/e2studio/RX65N/GR-ROSE/smc/smc.scfg:567]: Not sure you can/should fix these. This is a property for a code generation tool. Prefer you just add an exception for these files.

**Review by dgarske — CHANGES_REQUESTED** [review]
Otherwise looks good

**Comment by dgarske** [comment]
Retest this please:

PRB-known-configs.txt_70': FAIL: scripts/external
wolfSSL error: tcp connect failed: Connection timed out

**Review by dgarske — APPROVED** [review]

**Review by philljj — APPROVED** [review]

**Review by douzzer — APPROVED** [review]
Note, min version of `codespell` for a clean run is 2.3.0.  Before that, a bug made matching of lines in the `--exclude-file` sensitive to linebreak flavor, e.g. CRLF would throw it off.

**Pull request merged**